RFC/RFC-13-social-service.md

```diff
@@ -0,0 +1,158 @@
+# TITLE
+
+Author Name(s): @jmoguilevsky
+Author Date: 2022-10-06
+Due Date: 
+Status: `In progress`
+
+## Abstract
+
+<!--
+Insert here a brief paragraph summarizing the RFC in its current state.
+This section shall provide an overview of whether this is a settled
+decision, alternatives explored and a short summary of relevant
+background information and key insights.
+-->
+
+All current social features are implemented via abstractions on the client side, this poses a big problem when trying to add new features or apply business logic on top of features (i.e. max amount of friends) since with an API call any change can override any logic that the client has. This RFC proposes to have a new service on top of Matrix that will contain business logic, security policies, and will allow extracting the current abstractions from Matrix.
+
+## Need
+
+<!--
+Why is this RFC needed?  Briefly describe the need motivating this
+proposed artifact to be created or work be done.  What problem does it
+solve? Include an estimate of actual or perceived effort/gain.
+-->
+
+Whenever a new feature regarding social interaction between users is envisioned currently, Matrix is seen as a blocker (or the integration with it), for example there’s currently no way to know the friends of a certain user, the only user case currently supported is knowing a users’ own friends (when logged in) 
+
+Some features that can’t be correctly implemented with the current status of the feature:
+- Blocking another user (the message is still sent if a request to the server is done)
+- Knowing the friends of a user (e.g. for social recommendations)
+- Maxing out the channels or friends per user (preventing overusage of the chat)
+
+
+## Approach
+
+<!--
+How do you intend on addressing the need?  Describe what you plan on
+doing and the rationale behind the decisions you propose.  Then lay out
+the plan of execution, in rough order of how the execution should take
+place.  Include the rollout plan as well. (This is usually the longest
+section of the RFC) Hint: don’t be afraid of posting illustrations! The
+level of detail here has to be enough to give the reader  a clear
+understanding of the solution - it is up to the writer to decide.
+Further detail can be addressed to satisfy comments and increase clarity.
+-->
+
+We propose to have a new service that will maintain every current feature of the chat but will also have support for the social features, for chat features it will work as a proxy but will first apply any business logic designed by us.
```

Comment:
By "every" you literally mean "all the endpoints"?

Reply:
Except the endpoints that will be either replaced or proxied via some custom logic

Reply:
That way we can support backwards compatibility into a smooth transition